### DISCUSSION TITLE
Only Scale CPU and Ram not Storage

### DISCUSSION THREAD

**Discussion by JakobStadlhuber** [mysticaltech/terraform-hcloud-kube-hetzner#1344] [Ideas]
There is an option in hetzner when scaling servers to allow downscaling by not scaling the storage, it would be amazing if this would be possible too.
<img width="1405" alt="Bildschirmfoto 2024-05-09 um 15 27 46" src="https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/assets/20328243/bbc819dd-bd52-4ad9-a000-ef774e09b508">

Example with a possible thew param **scale_storage**

```
    {
      name        = "agent-small-x",
      server_type = "cax21",
      location    = "fsn1",
      scale_storage = false
      count       = 1,
      labels      = [],
      taints      = [],
      placement_group = "agent-group-fsn1-x1"  # Ensure this is unique
    }
```

**Comment by WebSpider** [comment]
I guess your use-case here is:
1. Define a nodepool, and have it spawned by terraform
2. Change the `server_type` in the nodepool
3. Now you want terraform to run by just replacing the VM-type, with the option to keep the disksize original (to support downsizing, or just to save storage costs with Hetzner)

Is that correct?

If so, there already is support in the `hcloud` terraform provider for this, the option is called `keep_disk`, so adding this should be relatively straightforward.

**Comment by WebSpider** [comment]
Make a small PR for this, let's see if I got it right the first time :D

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/pull/1347

**Comment by mysticaltech** [comment]
Implemented in PR: https://github.com/mysticaltech/terraform-hcloud-kube-hetzner/pull/2074

Closing this discussion as addressed for the v3 workstream.
